### PR TITLE
add command line interface for library

### PR DESCRIPTION
**Pull request by collinsmezie**
List all books.
List all people.
Create a person (teacher or student, not a plain Person).
Create a book.
Create a rental.
List all rentals for a given person id.

FUNCTIONS:
Present the user with a list of options to perform.
Lets users choose an option.
If needed, ask for parameters for the option.
Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_all_rentals_for_a_given_person_id
+    else
+      puts 'Thank you for using this App'
+      exit
+    end
+  end
```
Comment (danmainah) [inline app.rb]: - [ ] Kindly move these functions into your `main.rb` file, this is a project requirement.
- [ ] Also kindly ensure your app is functioning am getting this error from one of your files when I run it locally
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @collinsmezie , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good PR and Description


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts "7 - Exit\n\n"
+    gets.chomp
+  end
```
Comment (rukundoeric) [inline app.rb:25]: - [ ] <b>REQUIRED:</b> You still haven't fixed some issues raised from the previous review.
  - When I try to create a person with type teacher, the app throws an error:
    Screenshot(15)
  - When I try to create a person with student type, the app can't be able to set a permission. 
  Screenshot(16)

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @collinsmezie, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+  def create_person
+    print "\n Do you want to create a student (1) or a teacher (2)? [Input the number]: "
+    user_input = gets.chomp
+    case user_input
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
```
Comment (ShoiraTa) [inline app.rb:67]: - [ ] I still cannot add a teacher, I tried to check the reason, and found that you added create_student method but did not add create_teacher method. So when I select create_teacher method, app just cannot find the method to run. Please add create_teacher to make the app work. You can start from checking the code below:  


```
  def create_teacher
    puts 'Enter age: '
    age = gets.chomp
    puts 'Enter name: '
    name = gets.chomp
    puts "'Has parent permission? [Y/N]: '"
    permission = gets.chomp
      teacher = Teacher.new(age, name, parent_permission: false)
      @person.push(teacher)
  end

```

After adding the missing method, please make sure that it works properly (name, age, and etc fields are added in object)

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @collinsmezie ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Great commit history! ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
### **Hi @collinsmezie,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**